### PR TITLE
bump virt-manager/view-viewer and deps

### PR DESCRIPTION
**Pull request by jouve**

### CODE REVIEW

**Comment by doomedraven** [comment]
oh i was just starting to look to get latest virt-manager, can verify this works on mac os x 11.3 beta

**Review by odyssey4me — APPROVED** [review]

**Comment by fkpwolf** [comment]
I tried this pr and it can open virt-manager UI but when I open Spice view, I got error:
```
spice-client-error-quark: Could not auto-redirect USB Device at 20-22: could not claim interface 0(configuration 1): LIBUSB_ERROR_ACCESS(0)
```

File: virt-manager.rb [hunk 1]
```diff
@@ -3,57 +3,62 @@ class VirtManager < Formula
 
   desc "App for managing virtual machines"
   homepage "https://virt-manager.org/"
-  url "https://virt-manager.org/download/sources/virt-manager/virt-manager-2.2.1.tar.gz"
-  sha256 "cfd88d66e834513e067b4d3501217e21352fadb673103bacb9e646da9f029a1b"
-  revision 3
+  url "https://releases.pagure.org/virt-manager/virt-manager-3.2.0.tar.gz"
```
Comment (ritorujon) [inline virt-manager.rb:6]: why are you not using https://virt-manager.org/download/sources/  ?

Comment (jouve) [reply virt-manager.rb:6]: I guess I opened https://virt-manager.org/download/sources/virt-manager/ which redirects to release.pagure.org :)

**Comment by rafaelreis-r** [comment]
PR #166 on a Air M1

virt-viewer builds fine and works. 

virt-manager fails on:
```
==> /opt/homebrew/Cellar/virt-manager/3.2.0_3/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--PyGObject-2021
Last 15 lines from /Users/rafaelreisr/Library/Logs/Homebrew/virt-manager/07.pip:
  ERROR: Failed building wheel for PyGObject
Failed to build PyGObject
ERROR: Could not build wheels for PyGObject which use PEP 517 and cannot be installed directly
Exception information:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/virt-manager/3.2.0_3/libexec/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 180, in _main
    status = self.run(options, args)
  File "/opt/homebrew/Cellar/virt-manager/3.2.0_3/libexec/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 204, in wrapper
    return func(self, options, args)
  File "/opt/homebrew/Cellar/virt-manager/3.2.0_3/libexec/lib/python3.9/site-packages/pip/_internal/commands/install.py", line 360, in run
    raise InstallationError(
pip._internal.exceptions.InstallationError: Could not build wheels for PyGObject which use PEP 517 and cannot be installed directly
WARNING: You are using pip version 21.1.1; however, version 21.1.3 is available.
You should consider upgrading via the '/opt/homebrew/Cellar/virt-manager/3.2.0_3/libexec/bin/python3.9 -m pip install --upgrade pip' command.
Removed build tracker: '/private/tmp/pip-req-tracker-25m6ew6y'
```

I read somewhere that this seems to be an issue with that version of pip and M1 Macs. Version 21.1.3 works fine. I have updated pip to the latest version on home-brew, but don't know exactly how to do it on virt-manager build environment.

**Comment by jouve** [comment]
I can only test on catalina

**Comment by jouve** [comment]
@rafaelreis-r  can you try the following ?

```
curl -O https://files.pythonhosted.org/packages/51/2f/4d5d5afb7000b9151e33952b59163c9389bd867ac6fe85d62f85831fa061/PyGObject-3.40.1.tar.gz
tar xf PyGObject-3.40.1.tar.gz
cd PyGObject-3.40.1/
python3.9 -m venv .venv
./.venv/bin/pip install --upgrade pip==21.1.1
./.venv/bin/pip wheel -v .
```

**Comment by rafaelreis-r** [comment]
> @rafaelreis-r can you try the following ?

Commands worked fine, but build process still presents same error. 

```  Building wheel for PyGObject (PEP 517) ... done
  Created wheel for PyGObject: filename=PyGObject-3.40.1-cp39-cp39-macosx_11_0_arm64.whl size=236939 sha256=56d3126adf1682b0c9a82acbc655e8f7ce1fa48753c71d5967a8db246a5b74ab
  Stored in directory: /Users/rafaelreisr/Library/Caches/pip/wheels/22/64/50/c8c8c647d3bf799e350edaba27905aa28e0a093f0f9d2e9ac6
Successfully built PyGObject
WARNING: You are using pip version 21.1.1; however, version 21.1.3 is available.
You should consider upgrading via the '/opt/homebrew/Library/Taps/jeffreywildman/homebrew-virt-manager/PyGObject-3.40.1/.venv/bin/python3.9 -m pip install --upgrade pip' command.
Removed build tracker: '/private/var/folders/_1/41hvxkq93p71_s_g6_6d4mnr0000gn/T/pip-req-tracker-ugm2rck7' 
```
Installation:

```
==> Installing jeffreywildman/virt-manager/virt-manager
==> Patching
==> python3 -m venv --system-site-packages /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec
==> /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--certifi-20210715-14749-4ogwsh/certifi-2021.5.30
==> /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--charset-normalizer-20210715-14749-tufesi/charset-normalizer
==> /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--idna-20210715-14749-1dtint7/idna-3.2
==> /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--libvirt-python-20210715-14749-bpb72i/libvirt-python-7.5.0
==> /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--pycairo-20210715-14749-1etdh6y/pycairo-1.20.1
==> /opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /private/tmp/virt-manager--PyGObject-20210715-14749-10sxwp4/PyGObject-3.40.1
Last 15 lines from /Users/rafaelreisr/Library/Logs/Homebrew/virt-manager/07.pip:
  error: command '/opt/homebrew/Library/Homebrew/shims/mac/super/clang' failed with exit code 1
  Building wheel for PyGObject (PEP 517): finished with status 'error'
  ERROR: Failed building wheel for PyGObject
Failed to build PyGObject
ERROR: Could not build wheels for PyGObject which use PEP 517 and cannot be installed directly
Exception information:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 180, in _main
    status = self.run(options, args)
  File "/opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 205, in wrapper
    return func(self, options, args)
  File "/opt/homebrew/Cellar/virt-manager/3.2.0_4/libexec/lib/python3.9/site-packages/pip/_internal/commands/install.py", line 360, in run
    raise InstallationError(
pip._internal.exceptions.InstallationError: Could not build wheels for PyGObject which use PEP 517 and cannot be installed directly
Removed build tracker: '/private/tmp/pip-req-tracker-nhy48g95'

Do not report this issue to Homebrew/brew or Homebrew/core!
```

I noticed pyGobject is part of Homebrew already. I was able to go further by adding it to the rb file instead of compiling it in the install process. That compiles, but it leads me to a segmentation fault documented on this comment: https://github.com/jeffreywildman/homebrew-virt-manager/issues/173#issuecomment-866003100

**Comment by theranger** [comment]
I can confirm that using replacing PyGobject resource with `depends_on "pygobject3"` makes it possible to complete the installation on BigSur (Intel CPU). Application runs, connects to server, but opening any of the virtual instances throws Python memory management exception:

> Python(78266,0x10b117e00) malloc: *** error for object 0x7fb9c904e450: pointer being freed was not allocated
> Python(78266,0x10b117e00) malloc: *** set a breakpoint in malloc_error_break to debug

Have not yet managed to dig deeper into this.

**Comment by Damenly** [comment]
Well, thanks to Jouve, I fixed the pygobject installation issue in [my branch](https://github.com/Damenly/homebrew-virt-manager) but there is a segmentation fault issue: https://gitlab.gnome.org/GNOME/pygobject/-/issues/455
It seems a bug of pygobject but no one fixed it in the past 7 moths.

**Comment by jouve** [comment]
pypi and homebrew-core/pyobject3 are the same archive (3.40.1):

 * https://pypi.org/project/PyGObject/#files
 * https://github.com/Homebrew/homebrew-core/blob/master/Formula/pygobject3.rb 

there are not a lot of changes between 3.40.1 and master, so it might be the same bug :
https://gitlab.gnome.org/GNOME/pygobject/-/compare/3.40.1...master?from_project_id=416

**Comment by Damenly** [comment]
I just fixed segmentation faults with some hacks in my [branch](https://github.com/Damenly/homebrew-virt-manager).
Anyone who interests in it could look at [this](https://github.com/Damenly/pygobject/commit/01013b1003bac67061e595cd857be9cf7b69d7fd).

Tested it in arm64 only:
Screen Shot 2021-09-04 at 14 27 02

**Comment by rafaelreis-r** [comment]
@Damenly I can confirm your branch compiles and runs fine on my M1 Air! 

However I gave up on homebrew virt-manager for arm64 months ago since this hadn't had a solution all this time. 

In case anybody is interested I went the docker way using https://github.com/m-bers/docker-virt-manager on the host. Virt-manager on a webpage :)

**Comment by Krish-sysadmin** [comment]
@jouve  Should probably close this pull request. My pull request upgrades the libvirt-python version to a more suitable one.

**Comment by jouve** [comment]
https://github.com/Homebrew/homebrew-core/pull/91218/files

**Pull request merged**